servers/zms/src/main/java/com/yahoo/athenz/zms/DBService.java

```diff
@@ -2349,6 +2391,54 @@ Policy updateTemplatePolicy(Policy policy, String domainName, String policyName,
         return templatePolicy;
     }
     
+    ServiceIdentity updateTemplateServiceIdentity(ServiceIdentity serviceIdentity, String domainName, String serviceIdentityName, List<TemplateParam> params) {
+        ServiceIdentity templateServcieIdentity = new ServiceIdentity().setName(ZMSUtils.serviceResourceName(domainName, serviceIdentityName));
+        templateServcieIdentity.setDescription("");
+        if (serviceIdentity.getDescription() != null) {
+            templateServcieIdentity.setDescription(serviceIdentity.getDescription());
+        }
+        List<PublicKeyEntry> publicKeyEntries = serviceIdentity.getPublicKeys();
+        List<PublicKeyEntry> newPublicKeyEntries = new ArrayList<PublicKeyEntry>();
+        if (publicKeyEntries != null && !publicKeyEntries.isEmpty()) {
+            for (PublicKeyEntry publicKeyEntry : publicKeyEntries) {
+                PublicKeyEntry newPublicKeyEntry = new PublicKeyEntry();
+                newPublicKeyEntry.setId(publicKeyEntry.getId());
+                newPublicKeyEntry.setKey(publicKeyEntry.getKey());
+                newPublicKeyEntries.add(newPublicKeyEntry);
+            }
+        }
+        templateServcieIdentity.setPublicKeys(newPublicKeyEntries);
+        
+        List<String> hosts = serviceIdentity.getHosts();
+        List<String> newHosts = new ArrayList<String>();
+        
+        if (hosts != null && !hosts.isEmpty()) {
+            for (String host: hosts) {
+              newHosts.add(host);
+            }
+        }
+        
+        templateServcieIdentity.setHosts(newHosts);
+        
+        if (serviceIdentity.getExecutable() != null) {
```

Comment:
For the next 4 if blocks, we don't need the if blocks. We should just go head and set the values even if they're null directly:
templateServcieIdentity.setExecutable(serviceIdentity.getExecutable());
...